File: Reverse _Integer/revInt.cpp
class Solution {
public:
    int reverse(int x) {
        int num=x;
        int rev=0;
        while(num!=0){
           int rem= num % 10;

           if((rev < INT_MIN/10)|| (rev> (INT_MAX/10)) )
           {
            return 0;
           }
           rev= (rev*10) + rem;
           num=num/10;
        }
        return rev;
    }
};

File: Reverse _Integer/revInt.py
class Solution:
    def reverse(self, x: int) -> int:
        num=x
        rev=0
        while(num!=0):
            rem=int(num%10 if x>0 else abs(num)%10)
            rev=rev*10 + rem
            num=int(num/10)

            if( rev>pow(2,31) or rev<-(pow(2,31)-1) ):
                return 0
            # if( rev>(2^31) or rev<((2^31)-1) ):
            #     return 0
                
        return rev if x>0 else -rev
        